pkg/proxy/handlers/spacelister.go

```diff
@@ -0,0 +1,155 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/application"
+	"github.com/codeready-toolchain/registration-service/pkg/application/service"
+	"github.com/codeready-toolchain/registration-service/pkg/context"
+	"github.com/codeready-toolchain/registration-service/pkg/log"
+	"github.com/codeready-toolchain/registration-service/pkg/signup"
+	commonproxy "github.com/codeready-toolchain/toolchain-common/pkg/proxy"
+
+	"github.com/labstack/echo/v4"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+type SpaceLister struct {
+	GetSignupFunc          func(userID, username string) (*signup.Signup, error)
+	GetInformerServiceFunc func() service.InformerService
+}
+
+func NewSpaceLister(app application.Application) *SpaceLister {
+	return &SpaceLister{
+		GetSignupFunc:          app.SignupService().GetSignupFromInformer,
+		GetInformerServiceFunc: app.InformerService,
+	}
+}
+
+func (s *SpaceLister) HandleSpaceListRequest(ctx echo.Context) error {
+	userID, _ := ctx.Get(context.SubKey).(string)
+	username, _ := ctx.Get(context.UsernameKey).(string)
+	workspaceName := ctx.Param("workspace")
+	doGetWorkspace := len(workspaceName) > 0
+
+	signup, err := s.GetSignupFunc(userID, username)
+	if err != nil {
+		log.Error(nil, err, "error retrieving signup")
+		return errorResponse(ctx, apierrors.NewInternalError(fmt.Errorf("user account lookup failed")), http.StatusInternalServerError)
```

Comment:
I might be missing something but this is not currently covered by the tests, right?